app/models/admin_dashboard_data.rb

```diff
@@ -80,7 +84,8 @@ def self.add_problem_check(*syms, &blk)
     @@problem_blocks << blk if blk
   end
 
-  def self.add_scheduled_problem_check(check_identifier, &blk)
+  def self.add_scheduled_problem_check(check_identifier, klass = nil, &blk)
+    @@problem_scheduled_check_klasses[check_identifier] = klass
```

Comment:
I want to consolidate these two in a future clean-up to make checks first-class citizens and get rid of check methods in `AdminDashboardData`, but this is backwards compatible for now.